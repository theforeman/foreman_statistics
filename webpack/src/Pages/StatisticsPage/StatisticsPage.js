import React from 'react';
import PropTypes from 'prop-types';
import { translate as __ } from 'foremanReact/common/I18n';
import PageLayout from 'foremanReact/routes/common/PageLayout/PageLayout';
import Statistics from './Statistics/Statistics';

const StatisticsPage = ({ statisticsMeta, ...props }) => (
  <PageLayout header={__('Statistics')} searchable={false}>
    <Statistics statisticsMeta={statisticsMeta} {...props} />
  </PageLayout>
);

StatisticsPage.propTypes = {
  statisticsMeta: PropTypes.array,
};

StatisticsPage.defaultProps = {
  statisticsMeta: [],
};

export default StatisticsPage;
