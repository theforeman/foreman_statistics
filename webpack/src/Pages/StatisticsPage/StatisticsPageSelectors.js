export const selectStatisticsPage = state =>
  state.foremanStatistics.statisticsPage;

export const selectStatisticsMetadata = state =>
  selectStatisticsPage(state).metadata;
export const selectStatisticsIsLoading = state =>
  selectStatisticsPage(state).isLoading;
export const selectStatisticsMessage = state =>
  selectStatisticsPage(state).message;
export const selectStatisticsHasError = state =>
  selectStatisticsPage(state).hasError;
export const selectStatisticsHasMetadata = state =>
  selectStatisticsPage(state).hasData;
