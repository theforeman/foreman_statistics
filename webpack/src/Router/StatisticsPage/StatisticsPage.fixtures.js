import { noop } from 'foremanReact/common/helpers';
import { statisticsMeta } from '../../Components/StatisticsChartsList/StatisticsChartsList.fixtures';

export const statisticsProps = {
  statisticsMeta,
  isLoading: false,
  hasData: true,
  hasError: false,
  message: {},
  getStatisticsMeta: noop,
};
