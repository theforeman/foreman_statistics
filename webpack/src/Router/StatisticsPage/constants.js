export const STATISTICS_PAGE_DATA_RESOLVED = 'STATISTICS_PAGE_DATA_RESOLVED';
export const STATISTICS_PAGE_DATA_FAILED = 'STATISTICS_PAGE_FETCH_FAILED';
export const STATISTICS_PAGE_HIDE_LOADING = 'STATISTICS_PAGE_HIDE_LOADING';
export const STATISTICS_PAGE_URL = '/foreman_statistics/statistics';
