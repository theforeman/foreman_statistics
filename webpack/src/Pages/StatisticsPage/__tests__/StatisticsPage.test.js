import { testComponentSnapshotsWithFixtures } from '@theforeman/test';
import { statisticsProps } from '../StatisticsPage.fixtures';
import StatisticsPage from '../StatisticsPage';

const fixtures = {
  'render with props': statisticsProps,
};

describe('StatisticsPage', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(StatisticsPage, fixtures));
});
