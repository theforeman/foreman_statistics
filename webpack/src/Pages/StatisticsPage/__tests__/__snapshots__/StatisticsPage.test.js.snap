// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatisticsPage rendering render with props 1`] = `
<PageLayout
  header="Statistics"
  searchable={false}
>
  <Component
    getStatisticsMeta={[Function]}
    hasData={true}
    hasError={false}
    isLoading={false}
    message={Object {}}
    statisticsMeta={
      Array [
        Object {
          "id": "operatingsystem",
          "search": "/hosts?search=os_title=~VAL~",
          "title": "OS Distribution",
          "url": "statistics/operatingsystem",
        },
        Object {
          "id": "architecture",
          "search": "/hosts?search=facts.architecture=~VAL~",
          "title": "Architecture Distribution",
          "url": "statistics/architecture",
        },
      ]
    }
  />
</PageLayout>
`;
